spidermon/contrib/actions/telegram/__init__.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import absolute_import
+
+import ast
+import json
+import logging
+import requests
+
+import six
+
+from spidermon.contrib.actions.templates import ActionWithTemplates
+from spidermon.exceptions import NotConfigured
+
+logger = logging.getLogger(__name__)
+
+
+class SimplyTelegramClient:
+    send_message_api = "https://api.telegram.org/bot{token}/sendMessage?chat_id={chat_id}&parse_mode=Markdown&text={text}"
+
+    def __init__(self, token):
+        self.token = token
+
+    def send_message(self, message, recipient):
+        api_url = self.send_message_api.format(
+            token=self.token, chat_id=recipient, text=message
+        )
+        r = requests.get(api_url).json()
+        if r.get("ok") is False:
+            logger.error("Failed to send message. Telegram api error: %s", r.text)
+
+
+class TelegramMessageManager:
+    sender_token = None
+
+    def __init__(self, sender_token=None, fake=False):
+        sender_token = sender_token or self.sender_token
+        if not sender_token:
+            raise NotConfigured("You must provide a telegram bot token.")
+
+        self.fake = fake
+        self._client = SimplyTelegramClient(sender_token)
+
+    def send_message(self, to, text):
+        if self.fake:
+            logger.info(text)
+            return
+        if isinstance(to, list):
+            for recipient in to:
+                self._client.send_message(text, recipient)
+        else:
+            self._client.send_message(text, to)
+
+
+class SendTelegramMessage(ActionWithTemplates):
+    message = None
+    message_template = "telegram/default/message.jinja"
+    recipients = None
+    sender_token = None
+    include_message = True
+    fake = False
+
+    def __init__(
+        self,
+        sender_token=None,
+        recipients=None,
+        message=None,
+        message_template=None,
+        include_message=None,
+        fake=None,
+    ):
+        super(SendTelegramMessage, self).__init__()
+
+        self.fake = fake or self.fake
+        self.manager = TelegramMessageManager(
+            sender_token=sender_token or self.sender_token, fake=self.fake
+        )
+
+        self.recipients = recipients or self.recipients
+        self.message = message or self.message
+        self.message_template = message_template or self.message_template
+        self.include_message = (
+            include_message if include_message is not None else self.include_message
+        )
+        if not self.fake and not self.recipients:
```

Comment:
In my opinion, `recipients` must raise a `NotConfigured` error even if it is set to fake the response.
Usually we use fake just for development (or debugging in production), so we need to be sure that even in this case, the mandatory setting of recipients must be set.
Also, could you please add a test for this scenario?

Reply:
Done :heavy_check_mark: